pkgs/top-level/by-name-overlay.nix

```diff
@@ -31,9 +31,11 @@ let
       # Additionally in either of those alternatives, we would have to duplicate the hardcoding of "README.md"
       { }
     else
-      mapAttrs
-        (name: _: baseDirectory + "/${shard}/${name}/package.nix")
-        (readDir (baseDirectory + "/${shard}"));
+      lib.filterAttrs (_: path: builtins.pathExists path) (
+        mapAttrs (name: _: baseDirectory + "/${shard}/${name}/package.nix") (
+          readDir (baseDirectory + "/${shard}")
```

Comment:
Perhaps this is a use case for `lib.pipe`?

Reply:
That's bad for eval perf.